examples/nlp/language_modeling/megatron_change_num_partitions.py

```diff
@@ -1384,6 +1385,38 @@ def main():
                 with open_dict(model.cfg):
                     model.cfg.tokenizer.model = args.tokenizer_model_path
 
+            else:
+                # Extract all artifacts that are not binary files
+                # and save them to the model directory
+                logging.info("Extracting artifact from NeMo file...")
+                temp_dir = tempfile.mkdtemp()
+                tokenizer_model_path = None
+                with tarfile.open(args.model_file, "r") as tar:
+                    for member in tar.getmembers():
+                        if member.name == '.':
+                            continue
+
+                        if member.type not in ['ckpt', 'pt']:
+                            if 'mp_rank' in member.name:
+                                continue
+
+                            extracted_file = tar.extractfile(member)
+                            extracted_file_path = os.path.join(temp_dir, member.name)
+
+                            if '.model' in member.name:
+                                tokenizer_model_path = extracted_file_path
+
+                            with open(extracted_file_path, "wb") as f:
+                                f.write(extracted_file.read())
```

Comment:
Besides the tokenizer ".model" file, are there other files required by the tokenizer that need to be extracted? (at first glance it seems like we're not using any of the other extracted files)

Reply:
In that case, if only the ".model" file is needed, then why not only extract this file?

Reply:
There are models which dont have it, or have multiple. This is a fallback to not require the tokenizer all the time.

Reply:
Sorry, I'm not sure I understand:
* If there's no ".model" file, you may end up extracting a bunch of files that will never be used
* If there are multiple ".model" files, you end up using only the last one (which I guess is an arbitrary choice) and extracting a bunch of other files that will never be used

Why not just extract at most one ".model" file and stop there? (skipping over anything that's not ".model")

Reply:
This only extracts at the root directory, not at checkpoint level.  So the cost of extraction is trivial and order of kb or mb for huge Tokenizer. 

For either of those cases, if it has no Tokenizer it will fail when calling the constructor of the model - it always needs one tokenizer. For multiple tokenizer - this code path isn't supported at the moment. It's arbitrary yes, and will still fail when loading the constructor because such models require two Tokenizers and youre providing just one.

Reply:
Thanks for the clarification, but even if it's cheap, why extract multiple files when it's trivial to extract only the one you need? Note that (with my admittedly limited knowledge of NeMo) I see no reason why future models may not include very large files other than .ckpt or .pt...